casper/src/main/scala/coop/rchain/casper/Estimator.scala

```diff
@@ -69,17 +69,25 @@ object Estimator {
       newBlocks == blocks
 
     for {
-      scoresMap           <- buildScoresMap[F](blockDag)
-      sortedChildrenHash  = sortChildren(IndexedSeq(genesisBlockHash), blockDag.childMap, scoresMap)
+      scoresMap           <- buildScoresMap[F](blockDag, genesis)
+      sortedChildrenHash  = sortChildren(IndexedSeq(genesis.blockHash), blockDag.childMap, scoresMap)
       maybeSortedChildren <- sortedChildrenHash.toList.traverse(BlockStore[F].get)
       sortedChildren      = maybeSortedChildren.flatten.toVector
     } yield sortedChildren
   }
 
   // TODO: Fix to stop at genesis/LFB
-  def buildScoresMap[F[_]: Monad: BlockStore](blockDag: BlockDag): F[Map[BlockHash, Long]] = {
-    def hashParents(hash: BlockHash): F[List[BlockHash]] =
-      blockDag.dataLookup(hash).parents.pure[F]
+  def buildScoresMap[F[_]: Monad: BlockStore](
+      blockDag: BlockDag,
+      genesis: BlockMessage
+  ): F[Map[BlockHash, Long]] = {
+    def hashParents(hash: BlockHash, genesisBlockNumber: Long): F[List[BlockHash]] = {
```

Comment:
Looks like this doesn't need to return an `F` any more because everything is pure.